pkg/generate/cspc.go

```diff
@@ -193,32 +198,36 @@ func makePools(poolType string, nDevices int, bd map[string][]v1alpha1.BlockDevi
 				return nil, fmt.Errorf("not enough eligible blockdevices found on node %s, want %d, found %d", nodes[i], nDevices, len(bds))
 			}
 			index := 0
-			for d := 0; d < nDevices/2; d++ {
-				raids := []cstorv1.CStorPoolInstanceBlockDevice{{BlockDeviceName: bds[index].Name},
-					{BlockDeviceName: bds[index+1].Name}}
+			for d := 0; d < nDevices/min; d++ {
+				var raids []cstorv1.CStorPoolInstanceBlockDevice
+				for j := 0; j < min; j++ {
+					raids = append(raids, cstorv1.CStorPoolInstanceBlockDevice{BlockDeviceName: bds[index].Name})
+					index++
```

Comment:
I can just copy-paste the validation logic for `2^n +1` or `2^n +2` BDs per RaidGroup.

Reply:
Yes, You can do that

Reply:
I need to think how I'll implement stacking similarly sized BDS in one RaidGroup now. Sorting might be a good trivial solution for the same.

Reply:
Yes, we can do sort of BlockDevices based on their size in descending order and start picking by index but this will not fit in all cases let's say userA has the following bds of size {100, 50, 50, 20} and if he requested to create mirror pool then we need to pick {50, 50} size disk then only user can get pool size without much wastage. 

**How about doing it in the following way**?
- Since we have a list of eligible blockdevices on a host, we will form blodeviceList based on their size(Keep note of sizes in descending order).
- Now let's pick the bucket which satisfies `min` no.of blockdevices and add them to the pool. Once after adding them to the pool remove them from a bucket.

Corner case we need to handle:
- If we are unable to fill up the minimum no.of blockdevices per raid group then add unpicked blockdevices to the raidGroup based on the size in decresing order.

Reply:
Okay, I'll have to approximate and create storage size buckets to hold similar sizes together and see if the applied `nDevices` can be given in a CSPC.

Reply:
Yes